File: dotnet/DataFormatter/DataFormatter.Common/DataService.cs
using System;
using System.Collections.Generic;

namespace DataFormatter.Common
{
    public class DataService : IDataService
    {
        private bool _connectionStatus = false;

        public void OpenConnection()
        {
            if (_connectionStatus)
                throw new InvalidOperationException("Connection is already open");

            _connectionStatus = true;
        }

        public List<Item> ReadData()
        {
            if (!_connectionStatus)
            {
                throw new InvalidOperationException("Must open the connection first");
            }

            return CreateDummyItems();
        }

        public bool ExportData(String exportedData)
        {
            Console.WriteLine("Exporting following data to some file...");
            Console.WriteLine(exportedData);
            return true;
        }

        public void CloseConnection()
        {
            _connectionStatus = false;
        }

        public void NotifyJobComplete()
        {
            Console.WriteLine("Job is complete");
        }

        public static List<Item> CreateDummyItems()
        {

            return new List<Item>
            {
                new Item {Id="1",Title = "First Test item",Value= 100},
                new Item {Id="2",Title = "second Test item",Value= 210},
                new Item {Id="3",Title = "third Test item",Value= 200},
                new Item {Id="4",Title = "fourth Test item",Value= 300},
            };
        }

    }
}

File: java/dataformatter/src/main/java/com/cefalo/oopcourse/experiment/DataService.java
package com.cefalo.oopcourse.experiment;

import java.util.Arrays;
import java.util.List;

/**
 * A basic data service class.
 */
public class DataService {
    private boolean connectionStatus = false;

    public void openConnection() throws Exception {
        if (connectionStatus) throw new IllegalStateException("Connection is already open");

        connectionStatus = true;
    }

    public List<Item> readData() throws Exception {
        if (!connectionStatus) {
            throw new IllegalStateException("Must open the connection first");
        }

        return createDummyItems();
    }

    public boolean exportData(String exportedData) {
        System.out.println("Exporting following data to some file...");
        System.out.println(exportedData);
        return true;
    }

    public void closeConnection() {
        connectionStatus = false;
    }

    public void notifyJobComplete() {
        System.out.println("Job is complete");
    }

    static List<Item> createDummyItems() {
        return Arrays.asList(
                createItem("1", "First Test item", 100.0),
                createItem("2", "second Test item", 210.0),
                createItem("3", "third Test item", 200.0),
                createItem("4", "fourth Test item", 300.0)
        );
    }

    private static Item createItem(String id, String title, double value) {
        return new Item(id, title, value);
    }
}
